crm-app/src/store/reducers.js

```diff
@@ -0,0 +1,80 @@
+import C from '../constants';
+import storage from '../utils/localStorage';
+
+export const customer = (state = {}, action) => {
+	switch (action.type) {
+		case C.ADD_CUSTOMER: {
+			return {
+				name: action.name,
+				mail: action.mail,
+				phone: action.phone
+			};
+		}
+		case C.UPDATE_CUSTOMER: {
+			return {
+					...state,
+					name: action.name,
+					phone: action.phone
+			};
+		}
+		default:
+			return state;
+	}
+}
+
+export const customers = (state = [], action) => {
+	switch (action.type) {
+		case C.ADD_CUSTOMER:
+			const newCustomers = [
+				...state,
+				customer({}, action)
+			].sort(compareByName);
+			storage.setCustomers(newCustomers);
+			return newCustomers;
+		case C.UPDATE_CUSTOMER: {
+			const {index, name, phone} = action;
+
+			if (index < 0) { return state; }
+
+			const
+				info = state[index],
+				shouldUpdated = info.name !== name || info.phone !== phone;
+			if (shouldUpdated) {
+				const newCustomers = state.slice();
+				newCustomers.splice(index, 1, customer(info, action))
+							.sort(compareByName);
```

Comment:
Is it reasonable to use `newCustomers.sort()`?

Reply:
I'm not sure, but it looks reasonable. Let me know if you have a better idea.